.buildkite/pipeline.yml

```
@@ -255,3 +255,4 @@ steps:
       - build-debian-packages
       - build-github-release
     command: ".buildkite/steps/upload-release-steps.sh"
+    if: 'build.env("DRY_RUN") == "false" && build.branch =~ /^(main|.*-.*-stable)$$/'
```

Does anything set the `DRY_RUN` environment variable to `"false"`? I think `build.env("DRY_RUN")` will evaluate to `null` if the environment variable is not set. So, even if nothing sets it and the build is on the main branch, this step won't run, which I don't think is the intention.

```suggestion
    if: build.env("DRY_RUN") == "true" || build.branch =~ /^(main|.*-.*-stable)$$/
```
Is this what was intended? Then this step will run on the main or stable branches, or if `DRY_RUN` is set to `true` in the build env.

It's also clearer to me if there are no unnecessary quotes around the YAML.

---

Argh, too many negations. Yes, you're right - the negation of

```shell
[[ ${BUILDKITE_BRANCH}" != "main" && "${DRY_RUN}" != "true" && "${BUILDKITE_BRANCH}" != *-*-stable ]]
```
is 

```shell
[[ ${BUILDKITE_BRANCH}" == "main" || "${DRY_RUN}" == "true" || "${BUILDKITE_BRANCH}" == *-*-stable ]]
```